packages/xo-server/src/api/pool.js

```diff
@@ -162,40 +162,34 @@ getPatchesDifference.resolve = {
 
 // -------------------------------------------------------------------
 
-export async function mergeInto({ source, target, force }) {
-  const sourceHost = this.getObject(source.master)
-  const targetHost = this.getObject(target.master)
-
-  if (sourceHost.productBrand !== targetHost.productBrand) {
-    throw new Error(
-      `a ${sourceHost.productBrand} pool cannot be merged into a ${
-        targetHost.productBrand
-      } pool`
+export async function mergeInto({ source, sources, target, force }) {
+  if (sources !== undefined) {
+    await this.checkPermissions(
+      this.user.id,
+      sources.map(source => [source, 'administrate'])
     )
-  }
-
-  const counterDiff = this.getPatchesDifference(source.master, target.master)
-  if (counterDiff.length > 0) {
-    const targetXapi = this.getXapi(target)
-    await targetXapi.installPatches({
-      patches: await targetXapi.findPatches(counterDiff),
-    })
-  }
-
-  const diff = this.getPatchesDifference(target.master, source.master)
-  if (diff.length > 0) {
-    const sourceXapi = this.getXapi(source)
-    await sourceXapi.installPatches({
-      patches: await sourceXapi.findPatches(diff),
+    return this.mergeInto({
+      force,
+      sources,
+      target,
     })
   }
 
-  await this.mergeXenPools(source._xapiId, target._xapiId, force)
+  return this.mergeInto({
+    force,
+    sources: [source.id],
+    target,
+  })
```

Comment:
Your implementation should work but I think, it will be easier to maintain if the code only has one path:

```js
export async function mergeInto({ source, sources = [source], target, force }) {
```